### PR TITLE
improvements on tests of imagecodecs module

### PR DESCRIPTION
**Pull request by sturkmen72**
[WIP] trivial improvements on tests of imagecodecs module

### CODE REVIEW

**Comment by sturkmen72** [comment]
i was not intending to prepare a PR to merge. but now i thing the proposed changes will be useful.

please review and consider the file changes [baboon.ppm](https://github.com/sturkmen72/opencv_extra/commit/f5d5c89716a120ba1471a5d3e1bdffa8bf52638c) and [read.png](https://github.com/sturkmen72/opencv_extra/commit/00bf59f5674452e0a8f0cf1410d774d0ee1aee40) ( let me create a PR to opencv_extra if acceptable)

**Comment by asmorkalov** [comment]
@sturkmen72 What is your motivation of png -> ppm replacement?

**Comment by sturkmen72** [comment]
> @sturkmen72 What is your motivation of png -> ppm replacement?

1. i replaced baboon.ppm with correct one ( older was an unrelated image and probably not used in tests) using this new image in tests will be meaningful.
2. "#if defined(HAVE_PNG) || defined(HAVE_SPNG)" alredy exist in code and HAVE_PNG can be OFF in very rare cases.
3. a bit related "IT’S TIME TO RETIRE LENA FROM COMPUTER SCIENCE"

As a final word, the proposed changes are open to discussion. It may be unimportant. Let me do required changes according devs opinions.

**Comment by asmorkalov** [comment]
@mshabunin @opencv-alalek what is you opinion?

File: modules/imgcodecs/test/test_common.cpp [hunk 1]
```diff
@@ -12,7 +12,7 @@ Mat generateTestImageBGR_()
     Size sz(640, 480);
     Mat result(sz, CV_8UC3, Scalar::all(0));
 
-    const string fname = cvtest::findDataFile("../cv/shared/baboon.png");
+    const string fname = cvtest::findDataFile("../python/images/baboon.bmp");
```
Comment (mshabunin) [inline modules/imgcodecs/test/test_common.cpp:15]: I believe using `cv/shared` is less confusing than using image from `python/images`. Maybe we can copy that BMP to `cv/shared`?

File: modules/imgcodecs/test/test_grfmt.cpp [hunk 2]
```diff
@@ -87,11 +87,17 @@ const string all_images[] =
     "readwrite/uint16-mono2.dcm",
     "readwrite/uint8-rgb.dcm",
 #endif
+#if defined(HAVE_PNG) || defined(HAVE_SPNG)
```
Comment (mshabunin) [inline modules/imgcodecs/test/test_grfmt.cpp:90]: We have GDAL library integration which should support all these image types (PNG, JPEG, TIFF). Perhaps it should be accounted too, though I didn't try to test GDAL backend specifically (with no other backends).

File: modules/imgcodecs/test/test_jpeg.cpp [hunk 3]
```diff
@@ -106,7 +106,7 @@ TEST(Imgcodecs_Jpeg, encode_empty)
 TEST(Imgcodecs_Jpeg, encode_decode_progressive_jpeg)
 {
     cvtest::TS& ts = *cvtest::TS::ptr();
-    string input = string(ts.get_data_path()) + "../cv/shared/lena.png";
+    string input = string(ts.get_data_path()) + "../python/images/baboon.bmp";
```
Comment (mshabunin) [inline modules/imgcodecs/test/test_jpeg.cpp:109]: Same comment about `cv/shared` here and in other places.

**Comment by vpisarev** [comment]
I would choose beautiful Lena over less-than-beautiful baboon. Honestly, I don't see any real justification to make this change.

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**